packages/kbn-esql-validation-autocomplete/src/validation/validation.ts

```diff
@@ -432,56 +430,41 @@ function validateFunction(
         // few lines above
         return;
       }
-      // if the arg is an array of values, check each element
-      if (Array.isArray(outerArg) && isArrayType(argDef.type)) {
-        const extractedType = extractSingleType(argDef.type);
-        const everyArgInListMessages = outerArg
-          .map((arg) => {
-            return [
-              validateFunctionLiteralArg,
-              validateNestedFunctionArg,
-              validateFunctionColumnArg,
-            ].flatMap((validateFn) => {
-              return validateFn(
-                astFunction,
-                arg,
-                { ...argDef, type: extractedType },
-                references,
-                parentCommand
-              );
-            });
-          })
-          .filter((ms) => ms.length);
-        if (everyArgInListMessages.length) {
-          failingSignature.push(
-            getMessageFromId({
-              messageId: 'wrongArgumentType',
-              values: {
-                name: astFunction.name,
-                argType: argDef.type,
-                value: `(${getAllArrayValues(outerArg).join(', ')})`,
-                givenType: `(${getAllArrayTypes(outerArg, parentCommand, references).join(', ')})`,
-              },
-              locations: {
-                min: (outerArg[0] as ESQLSingleAstItem).location.min,
-                max: (outerArg[outerArg.length - 1] as ESQLSingleAstItem).location.max,
-              },
-            })
-          );
-        }
-        return;
-      }
-      const wrappedArg = Array.isArray(outerArg) ? outerArg : [outerArg];
-      for (const actualArg of wrappedArg) {
-        const argValidationMessages = [
-          validateFunctionLiteralArg,
-          validateNestedFunctionArg,
-          validateFunctionColumnArg,
-        ].flatMap((validateFn) => {
-          return validateFn(astFunction, actualArg, argDef, references, parentCommand);
-        });
-        failingSignature.push(...argValidationMessages);
+
+      // check every element of the argument (may be an array of elements, or may be a single element)
+      const hasMultipleElements = Array.isArray(outerArg);
+      const argElements = hasMultipleElements ? outerArg : [outerArg];
+      const singularType = extractSingularType(argDef.type);
+      let messagesFromAllArgElements = argElements
+        .map((arg) => {
```

Comment:
```suggestion
        .flatMap((arg) => {
```

and remove the `flat` at the bottom?

Reply:
Done in https://github.com/elastic/kibana/pull/180625/commits/fc8ef2897360f85893c5d7288a917fb0421b7522